plugins/transformer/basic/client/spec/spec.go

```diff
@@ -31,42 +38,56 @@ func (s *Spec) SetDefaults() {
 	}
 }
 
+var kindToRequiredFields = map[string]map[string]struct{}{
+	KindRemoveColumns:    {"Columns": {}},
+	KindAddColumn:        {"Name": {}, "Value": {}},
+	KindObfuscateColumns: {"Columns": {}},
+	KindChangeTableNames: {"NewTableNameTemplate": {}},
+}
+
+var fieldsToCheck = []string{"Columns", "Name", "Value", "FromTableName", "NewTableNameTemplate"}
+
 func (s *Spec) Validate() error {
 	var err error
 	for _, t := range s.TransformationSpecs {
-		switch t.Kind {
-		case KindRemoveColumns:
-			if len(t.Columns) == 0 {
-				err = errors.Join(err, fmt.Errorf("'columns' field must be specified for remove_columns transformation"))
-			}
-			if t.Name != "" {
-				err = errors.Join(err, fmt.Errorf("'name' field must not be specified for remove_columns transformation"))
+		requiredFields, ok := kindToRequiredFields[t.Kind]
```

Comment:
Non blocking, but what's the reason to use reflection? I'm worried this can break when someone renames a field

Reply:
Can you think of an elegant way of doing this without reflection? If someone renames a field this will panic in the tests.

Reply:
I would keep the `switch/case` actually as it's easier to follow (and also not more code).

But if we want a generic way maybe a more declarative approach like https://pkg.go.dev/github.com/go-playground/validator/v10#section-readme (it uses reflection under the hood too I think)

Reply:
Again it's non blocking I think

Reply:
Reverted to switch/case.